code/modules/organs/internal/lungs.dm

```diff
@@ -87,11 +92,17 @@
 		else
 			owner.emote(pick("shiver","twitch"))
 
-	if(damage || world.time > last_successful_breath + 2 MINUTES)
-		owner.adjustOxyLoss(HUMAN_MAX_OXYLOSS*breath_fail_ratio)
-
+	owner.adjustOxyLoss(HUMAN_MAX_OXYLOSS * breath_fail_ratio)
 	owner.oxygen_alert = max(owner.oxygen_alert, 2)
 
+/obj/item/organ/internal/lungs/proc/enable_rupture()
+	rupture_imminent = TRUE
+	checking_rupture = TRUE
+	addtimer(CALLBACK(src, .proc/disable_rupture), 50)
```

Comment:
```suggestion
	addtimer(CALLBACK(src, .proc/disable_rupture), 5 SECONDS, TIMER_UNIQUE)
```

I think you need TIMER_UNIQUE on this timer and the enable_rupture timer or things can get wonky, or just create a bunch of disable_rupture timers. Also time macros are nice.